config/routes.rb

```diff
@@ -99,6 +99,7 @@
     get 'registrations/psych-sheet/:event_id' => 'registrations#psych_sheet_event', as: :psych_sheet_event
     resources :registrations, only: [:index, :update, :create, :edit, :destroy], shallow: true do
       get 'payments' => 'payment#registration_payments', as: :payments
+      get 'history' => 'registrations#registration_history', on: :member, as: :history
```

Comment:
what exactly does on :member do? I looked it up, but I don't immediately get why it's necessary here

Reply:
Simply put, it distinguishes between calling the route parameter `/registrations/:id` and `/registrations/:registration_id`. And the existing controller structure relies a lot on `:id`, so that's what I had to go with.

Unfortunately, this also prompts Rails to change the name of the route handler into some obscure prefix thing, I'm currently trying to find a solution for that,

Reply:
That won't hurt. But other places throughout the controller which perform pre-checks will explode if it's called `registration_id` instead of `id`